ompi/mca/io/romio321/romio/adio/common/ad_read_coll.c

```diff
@@ -864,9 +864,13 @@ static void ADIOI_R_Exchange_data(ADIO_File fd, void *buf, ADIOI_Flatlist_node
 	}
     }
 
-    statuses = (MPI_Status *) ADIOI_Malloc((nprocs_send+nprocs_recv+1) * \
+#ifdef MPI_STATUSES_IGNORE
+    statuses = MPI_STATUSES_IGNORE;
+#else
+    statuses = (MPI_Status *) ADIOI_Malloc((nprocs_send+nprocs_recv+1) *
                                      sizeof(MPI_Status)); 
-     /* +1 to avoid a 0-size malloc */
+    /* +1 to avoid a 0-size malloc */
+#endif
```

Comment:
This new stuff about MPI_STATUSES_IGNORE isn't really necessary.

Indeed, Open MPI has MPI_STATUSES_IGNORE, but it's not a #define, so it won't even pass the #ifdef, above.